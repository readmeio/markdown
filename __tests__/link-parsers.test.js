const markdown = require('../index');

test('a link with label', () => {
  expect(markdown.mdast('[link](http://www.foo.com)')).toMatchSnapshot();
});

test('a link ref', () => {
  expect(markdown.mdast('[link]')).toMatchSnapshot();
});

test('a bracketed autoLinked url', () => {
  expect(markdown.mdast('<http://www.google.com>')).toMatchSnapshot();
});

test('a bare autoLinked url', () => {
  expect(markdown.mdast('http://www.googl.com')).toMatchSnapshot();
});

test.skip('a bare autoLinked url with no protocol', () => {
  expect(markdown.mdast('www.google.com')).toMatchSnapshot();
});
