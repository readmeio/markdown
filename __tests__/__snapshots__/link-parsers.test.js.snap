// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a bare autoLinked url 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "http://www.googl.com",
            },
          ],
          "title": null,
          "type": "link",
          "url": "http://www.googl.com",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`a bare autoLinked url with no protocol 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "www.google.com",
            },
          ],
          "title": null,
          "type": "link",
          "url": "http://www.google.com",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`a bracketed autoLinked url 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "http://www.google.com",
            },
          ],
          "title": null,
          "type": "link",
          "url": "http://www.google.com",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`a link ref 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "link",
            },
          ],
          "identifier": "link",
          "label": "link",
          "referenceType": "shortcut",
          "type": "linkReference",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;

exports[`a link with label 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "type": "text",
              "value": "link",
            },
          ],
          "title": null,
          "type": "link",
          "url": "http://www.foo.com",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "root",
}
`;
